services/resource-usage-tracker/src/simcore_service_resource_usage_tracker/services/modules/db/licensed_items_checkouts_db.py

```diff
@@ -0,0 +1,215 @@
+from datetime import datetime
+from typing import cast
+
+import sqlalchemy as sa
+from models_library.licensed_items import LicensedItemID
+from models_library.products import ProductName
+from models_library.resource_tracker_licensed_items_checkouts import (
+    LicensedItemCheckoutID,
+)
+from models_library.rest_ordering import OrderBy, OrderDirection
+from models_library.wallets import WalletID
+from pydantic import NonNegativeInt
+from servicelib.rabbitmq.rpc_interfaces.resource_usage_tracker.errors import (
+    LicensedItemCheckoutNotFoundError,
+)
+from simcore_postgres_database.models.resource_tracker_licensed_items_checkouts import (
+    resource_tracker_licensed_items_checkouts,
+)
+from simcore_postgres_database.utils_repos import (
+    pass_or_acquire_connection,
+    transaction_context,
+)
+from sqlalchemy.ext.asyncio import AsyncConnection, AsyncEngine
+
+from ....models.licensed_items_checkouts import (
+    CreateLicensedItemCheckoutDB,
+    LicensedItemCheckoutDB,
+)
+
+_SELECTION_ARGS = (
+    resource_tracker_licensed_items_checkouts.c.licensed_item_checkout_id,
+    resource_tracker_licensed_items_checkouts.c.licensed_item_id,
+    resource_tracker_licensed_items_checkouts.c.wallet_id,
+    resource_tracker_licensed_items_checkouts.c.user_id,
+    resource_tracker_licensed_items_checkouts.c.user_email,
+    resource_tracker_licensed_items_checkouts.c.product_name,
+    resource_tracker_licensed_items_checkouts.c.service_run_id,
+    resource_tracker_licensed_items_checkouts.c.started_at,
+    resource_tracker_licensed_items_checkouts.c.stopped_at,
+    resource_tracker_licensed_items_checkouts.c.num_of_seats,
+    resource_tracker_licensed_items_checkouts.c.modified,
+)
+
+assert set(LicensedItemCheckoutDB.model_fields) == {
+    c.name for c in _SELECTION_ARGS
+}  # nosec
+
+
+async def create(
+    engine: AsyncEngine,
+    connection: AsyncConnection | None = None,
+    *,
+    data: CreateLicensedItemCheckoutDB,
+) -> LicensedItemCheckoutDB:
+    async with transaction_context(engine, connection) as conn:
+        result = await conn.execute(
+            resource_tracker_licensed_items_checkouts.insert()
+            .values(
+                licensed_item_id=data.licensed_item_id,
+                wallet_id=data.wallet_id,
+                user_id=data.user_id,
+                user_email=data.user_email,
+                product_name=data.product_name,
+                service_run_id=data.service_run_id,
+                started_at=data.started_at,
+                stopped_at=None,
+                num_of_seats=data.num_of_seats,
+                modified=sa.func.now(),
+            )
+            .returning(*_SELECTION_ARGS)
+        )
+        row = result.first()
+        return LicensedItemCheckoutDB.model_validate(row)
+
+
+async def list_(
```

Comment:
what is the `_` in suffix supposed to mean here?

Reply:
`list` is already an occupied Python function; therefore, this is a "Pythonic" way to use a function name that is already used by Python.